TIcTacToe/tic-tac-toe/src/components/Board.js

```diff
@@ -1,73 +1,78 @@
 import React, { useContext } from "react";
-import { GameContext } from "../contexts/GameContext";
-import Square from "./Square";
+import {
+  GameContext,
+  usePlayUserTurn,
+  useHistory,
+} from "../contexts/GameContext";
+import Square from "./Square.js";
 
-const Board = ({ squares }) => {
-  const context = useContext(GameContext);
+const Board = () => {
+  const playerTurn = usePlayUserTurn();
+  const history = useHistory();
+  const { stepNumber } = useContext(GameContext);
   return (
     <>
       <div className="board-row">
         <Square
-          value={squares[0]}
+          value={history[stepNumber][0]}
```

Comment:
`Square` can get these values directly from the context (both `playerTurn` and `value`), if you pass its index as a prop.

Reply:
use `usePlayUserTurn` hook  inside of `Square`. you only need to get the index of the square to be able to call function